client/app/components/form/widgets/MultiSelectWidget.tsx

```diff
@@ -0,0 +1,103 @@
+"use client";
+
+import { Autocomplete, Chip, MenuItem, TextField } from "@mui/material";
+import { WidgetProps } from "@rjsf/utils/lib/types";
+import { DARK_GREY_BG_COLOR, BC_GOV_SEMANTICS_RED } from "@/app/styles/colors";
+
+interface Option {
+  const: string | number;
+  title: string | number;
+}
+
+const MultiSelectWidget: React.FC<WidgetProps> = ({
+  id,
+  onChange,
+  rawErrors,
+  schema,
+  value,
+  uiSchema,
+}) => {
+  const fieldSchema = schema.items as {
+    enum: Array<string | number>;
+    enumNames?: Array<string | number>;
+    type: string;
+  };
+
+  // Using enum and enumNames as anyOf was triggering a lot of validation errors for multiselect
+  // If no enumNames are provided, use the enum values as the names. The enumNames will not be saved in the formData.
+  const enumValues = fieldSchema?.enum;
+  const enumNames = fieldSchema?.enumNames;
+  const options = enumValues.map((val: string | number, index: number) => ({
+    const: val,
+    title: enumNames?.[index] || val,
+  }));
+
+  const handleChange = (e: React.ChangeEvent<{}>, option: Array<Option>) => {
+    onChange(option.map((o: Option) => o.const));
+  };
+
+  const placeholder = uiSchema?.["ui:placeholder"]
+    ? `${uiSchema["ui:placeholder"]}...`
+    : "";
+
+  const displayPlaceholder = value?.length === 0;
+
+  const isError = rawErrors && rawErrors.length > 0;
+  const borderColor = isError ? BC_GOV_SEMANTICS_RED : DARK_GREY_BG_COLOR;
+
+  const styles = {
+    width: "100%",
+    "& .MuiOutlinedInput-notchedOutline": {
+      borderColor: borderColor,
+    },
+  };
+
+  return (
+    <Autocomplete
+      disablePortal
+      id={id}
+      multiple
+      filterSelectedOptions
+      autoHighlight
+      options={options}
+      value={value.map((val: string | number) => {
```

Comment:
I think `val` makes sense as a map variable name here because the RJSF prop really is called `value`. I see `val` used elsewhere too--if in any of those cases it's not RJSF's `value`, then I'd suggest changing it to something else to avoid confusion

Reply:
Sounds good, I have changed most of the other `val` names with one exception, right below this is the prop `isOptionEqualToValue` and I thought it made sense to keep `val` there as well:

```
  isOptionEqualToValue={(option: Option, val: Option) => {
        return option.id === val.id;
      }}
```

Let me know if you think otherwise.